### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by Page-**
Add linting and fix tests for node 12

### CODE REVIEW

File: tests/rindle.spec.js [hunk 1]
```diff
@@ -33,499 +34,526 @@ var EventEmitter = require('events').EventEmitter;
 var tmp = require('tmp');
 tmp.setGracefulCleanup();
 
-var rindle = require('../lib/rindle');
+chai.use(require('chai-as-promised'));
+chai.use(require('sinon-chai'));
 
-describe('Rindle:', function() {
-  'use strict';
+var { expect } = chai;
```
Comment (Page-) [inline tests/rindle.spec.js:40]: This was what was necessary for linting, however I also removed mochainon in the process of working that out and decided it made sense to keep it removed

File: lib/rindle.js [hunk 2]
```diff
@@ -104,18 +103,19 @@ exports.wait = function(stream, callback) {
...
+
+	stream.on('data', function (chunk) {
+		chunks += chunk;
+	});
+
+	return exports
+		.wait(stream)
+		.then(function () {
+			return chunks;
+		})
```
Comment (dfunckt) [inline lib/rindle.js:117]: Worth changing as follows, but feel free to ignore given it wasn't like that before and it's reasonable to expect we'll switch to native promises at some point:

```suggestion
		.return(chunks)
```

File: lib/rindle.js [hunk 3]
```diff
@@ -146,17 +146,14 @@ exports.extract = function(stream, callback) {
...
-  // Fulfil once we finish writing to both output streams
-  return Promise.all([
-    exports.wait(output1),
-    exports.wait(output2)
-  ]).nodeify(callback);
+exports.bifurcate = function (stream, output1, output2, callback) {
+	stream.pipe(output1);
+	stream.pipe(output2);
+
+	// Fulfil once we finish writing to both output streams
```
Comment (dfunckt) [inline lib/rindle.js]:
```suggestion
	// Fulfill once we finish writing to both output streams
```

**Review by dfunckt — APPROVED** [review]

**Review by sradevski — APPROVED** [review]

File: lib/rindle.js (hunk 2 above)
Comment (Page-) [reply lib/rindle.js:117]: Yeah my assumption would be we'll go to native promises and async/await at some point so I'd avoid adding bluebird sugar - utility functions are fair game imo though

File: lib/rindle.js (hunk 3 above)
Comment (Page-) [reply lib/rindle.js]: Technically both are used depending on flavor of English, I think you're right that fulfill is the more common (only?) one in American English though

Comment (dfunckt) [reply lib/rindle.js]: TIL! I was previously only aware of the "fulfill" spelling. From my computer's (British English) dictionary:

> fulfil | fʊlˈfɪl | (US fulfill)

**Pull request merged**